src/Dashboard/Resources/Filter/Filter.tsx

```diff
@@ -1,99 +1,114 @@
-import React, { useState, useEffect } from 'react';
+import { useState, useEffect } from 'react';
 
 interface Props {
-  onUpdate: (tags: string[]) => void;
+  onUpdate: (tags: string[], count: {'unit': number, 'type': number}) => void;
 }
 
-const Filter = ({ onUpdate }: Props) => {
+let prevTags = [''];
 
+const Filter = ({ onUpdate }: Props) => {
   const [selectedTags, setSelectedTags] = useState(['']);
+  const [selectedTagsCount, setSelectedTagsCount] = useState({'unit': 0, 'type': 0})
 
-  const handleTagSelect = (tag: string) => {
+  const handleTagSelect = (tag: string, tagClass: string|null) => {
     const isTagSelected = selectedTags.includes(tag);
 
     if (isTagSelected) {
       setSelectedTags(selectedTags.filter((t) => t !== tag));
+      if (tagClass === 'unit' || tagClass === 'type') {
+        let newTagCount = selectedTagsCount;
+        newTagCount[tagClass] -= 1;
+        setSelectedTagsCount(newTagCount);
+      }
     } else {
       setSelectedTags([...selectedTags, tag]);
+      if (tagClass === 'unit' || tagClass === 'type') {
+        let newTagCount = selectedTagsCount;
+        newTagCount[tagClass] += 1;
+        setSelectedTagsCount(newTagCount);
+      }
     }
   };
 
   useEffect(() => {
-    onUpdate(selectedTags);
+    if (prevTags !== selectedTags) {
+      prevTags = selectedTags;
+      onUpdate(selectedTags, selectedTagsCount);
+    }
   });
 
   return (
```

Comment:
Can you use a `useMemo` and `useEffect` hook to compute which filters we need to add here instead of putting them in manually? That way as we add more units, more resource types, we won't need to edit this file.

Reply:
not sure what this means but I edited files so that the only hardcoded tagNames is in the Resources.ts file as variables

```
interface Resource {
  title: string;
  description: string;
  img: string;
  subLinks: ResourceSubLink[];
  tags: {
    unit: string[],
    type: string[],
    misc: string[]};
  rating?: number;
}

interface ResourceSubLink {
  name: string;
  url: string;
}

const tagNames = ['unit', 'type', 'misc']
```